test/adapter/json/collection_test.rb

```diff
@@ -84,6 +84,72 @@ def test_include_option
 
           assert_equal(expected, actual)
         end
+
+        def test_fields_option
+          serializer = ActiveModel::Serializer::CollectionSerializer.new([@first_post, @second_post])
+          adapter = ActiveModelSerializers::Adapter::Json.new(serializer)
+          actual = adapter.serializable_hash(fields: [:id])
```

Comment:
Also, since `fields` is part of `JSON API`, we need to make sure fields behave properly there as well. (and the adapter already handles fields)